src/test/java/com/iexec/worker/feign/LoginServiceTests.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2022 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.worker.feign;
+
+import com.iexec.common.security.Signature;
+import com.iexec.common.utils.SignatureUtils;
+import com.iexec.worker.chain.CredentialsService;
+import com.iexec.worker.feign.client.CoreClient;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.http.ResponseEntity;
+import org.web3j.crypto.Credentials;
+import org.web3j.crypto.ECKeyPair;
+import org.web3j.crypto.Keys;
+
+import static com.iexec.worker.feign.LoginService.TOKEN_PREFIX;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class LoginServiceTests {
+
+    @Mock
+    CoreClient coreClient;
+    @Mock
+    CredentialsService credentialsService;
+    @InjectMocks
+    private LoginService loginService;
+
+    @BeforeEach
+    void init() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    @SneakyThrows
+    private Credentials generateCredentials() {
+        ECKeyPair ecKeyPair = Keys.createEcKeyPair();
+        return Credentials.create(ecKeyPair);
+    }
+
+    @Test
+    void shouldNotLoginOnEmptyChallenge() {
+        Credentials credentials = generateCredentials();
+        when(credentialsService.getCredentials()).thenReturn(credentials);
+        when(coreClient.getChallenge(credentials.getAddress())).thenReturn(ResponseEntity.ok(null));
+        assertAll(
+                () -> assertEquals("", loginService.login()),
+                () -> verify(coreClient).getChallenge(credentials.getAddress())
+        );
+    }
```

Comment:
I usually like to test both null and empty values with a parameterized test:
```suggestion
    @ParameterizedTest
    @NullSource
    @ValueSource(strings = "")
    void shouldNotLoginOnEmptyChallenge(String emptyChallenge) {
        Credentials credentials = generateCredentials();
        when(credentialsService.getCredentials()).thenReturn(credentials);
        when(coreClient.getChallenge(credentials.getAddress())).thenReturn(ResponseEntity.ok(emptyChallenge));
        assertAll(
                () -> assertEquals("", loginService.login()),
                () -> verify(coreClient).getChallenge(credentials.getAddress())
        );
    }
```

Reply:
See 092c9f4

Reply:
See 6b62123